### ISSUE TITLE
`import-error` when importing another module from same namespace

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#6538]
### Bug description

After #6405, linting `b.py` below raises `import-error`, so long as the parent of `test` is on `sys.path`:

```
test
    a.py
    b.py
```
**a.py**
```python
"""This module is a"""
print("You are importing a")
```
**b.py**
```python
"""This module imports a"""
import a
```

### Command used

```shell
pylint test/b.py
--or (no difference)--
cd test
pylint b.py
```

### Pylint output

```shell
************* Module b
test/b.py:2:0: E0401: Unable to import 'a' (import-error)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)
```

### Expected behavior

No msg, since `b.py` executes fine:
```shell
$ python3 b.py
You are importing a
```

### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.12.0-dev0
Python 3.10.1 (v3.10.1:2cd268a3a9, Dec  6 2021, 14:28:59) [Clang 13.0.0 (clang-1300.0.29.3)]
```

**Comment by DanielNoord** [comment]
Hm, we might want to revert #6405 so we can unblock the `beta`. I'd rather release `2.14` with the `argparse` changes and delay the `namespace` changes (again) than having to fiddle with this for another two weeks and block the release.

I saw we had over 400 items already on the milestone so it's probably best to release soon.

**Comment by Pierre-Sassoulas** [comment]
Opened https://github.com/PyCQA/pylint/pull/6548 to revert.

**Comment by jacobtylerwalls** [comment]
What makes this a little tricky to reproduce is that the parent of the namespace directory has to be on `sys.path`.

**Comment by jacobtylerwalls** [comment]
The test isolation issue with the regression test doesn't need block the beta.

**Comment by jacobtylerwalls** [comment]
Fixed in #6548 and tested in #6553
